web/src/Network.jsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useState } from "react";
+import { Stack, StackItem } from "@patternfly/react-core";
+import { useInstallerClient } from "./context/installer";
+import { useCancellablePromise } from "./utils";
+import { CONNECTION_TYPES } from "./client/network";
+import NetworkWiredStatus from "./NetworkWiredStatus";
+import NetworkWifiStatus from "./NetworkWifiStatus";
+
+export default function Network() {
+  const client = useInstallerClient();
+  const { cancellablePromise } = useCancellablePromise();
+  const [connections, setConnections] = useState([]);
+
+  useEffect(() => {
+    cancellablePromise(client.network.activeConnections()).then(setConnections);
+  }, [client.network, cancellablePromise]);
+
+  useEffect(() => {
+    const onConnectionAdded = addedConnection => {
+      setConnections(conns => [...conns, addedConnection]);
+    };
+
+    return client.network.listen("connectionAdded", onConnectionAdded);
+  }, [client.network]);
+
+  useEffect(() => {
+    const onConnectionRemoved = connectionPath => {
+      setConnections(conns => conns.filter(c => c.path !== connectionPath));
+    };
+
+    return client.network.listen("connectionRemoved", onConnectionRemoved);
+  }, [client.network]);
+
+  useEffect(() => {
+    const onConnectionUpdated = connection => {
+      setConnections(conns => {
+        const newConnections = conns.filter(c => c.path !== connection.path);
+        return [...newConnections, connection];
+      });
+    };
+
+    return client.network.listen("connectionUpdated", onConnectionUpdated);
+  }, [client.network]);
+
+  if (!connections.length) return null;
+
+  const activeWiredConnections = connections.filter(c => c.type === CONNECTION_TYPES.ETHERNET);
+  const activeWifiConnections = connections.filter(c => c.type === CONNECTION_TYPES.WIFI);
+
+  return (
```

Comment:
When there are none active connection

* Return an explanatory message (i.e., "Network cannot be configured", "No network connections available"), or
* Display a list of devices that can be configured to have an active connection

Something to do in next iterations, after having a more clear vision about the desired UI for that _Network Overview_ component (please, try to avoid making it too complex; we could have a _Network Section_ component for an advanced network setup).